modules/dnn/src/net.cpp

```diff
@@ -13,6 +13,7 @@ CV__DNN_INLINE_NS_BEGIN
 Net::Net()
     : impl(makePtr<Net::Impl>())
 {
+    setPreferableBackend(getParam_DNN_BACKEND_DEFAULT());
```

Comment:
I suggest to do the following:
on net_impl.cpp:
```
preferableBackend = DNN_BACKEND_DEFAULT;
```
on Net::Net():
```
setPreferableBackend(DNN_BACKEND_DEFAULT);
```

Existing code will do the rest:
https://github.com/opencv/opencv/blob/396a75f2eaab80f9a7d1a184b138bf179bf84cbd/modules/dnn/src/net_impl_backend.cpp#L167-L168